### PR TITLE
Refactor/permissions part 3

### PR DESCRIPTION
**Pull request by mimse**

### CODE REVIEW

File: libs/dh/admin/feature-permissions-new/src/lib/download.component.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  exportAsCsv(): void {
+    const basePath = 'admin.userManagement.permissionsTab.';
+    const headers = [
+      `"${translate(basePath + 'permissionName')}"`,
+      `"${translate(basePath + 'permissionDescription')}"`,
+    ];
+
+    const lines = this.permissions().map((x) => [`"${x.name}"`, `"${x.description}"`]);
+
+    exportToCSV({ headers, lines });
```
Comment (dzhavat) [inline libs/dh/admin/feature-permissions-new/src/lib/download.component.ts:65]: suggestion: let's give the download file a better name than the default.

```suggestion
    exportToCSV({ headers, lines, fileName: 'DataHub-Permissions' });
```

File: libs/dh/admin/feature-permissions-new/src/lib/download.component.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  downloadRelationCSV(url: string) {
+    this.toastService.open({
+      type: 'loading',
+      message: translate('shared.downloadStart'),
+    });
+
+    const fileOptions = {
+      name: 'permissions-relation-report',
```
Comment (dzhavat) [inline libs/dh/admin/feature-permissions-new/src/lib/download.component.ts:75]:
```suggestion
      name: 'DataHub-Permissions-relation-report',
```

File: libs/dh/admin/feature-permissions-new/src/lib/table.component.ts [hunk 3]
```diff
@@ -58,58 +48,40 @@ import { DhPermissionRequiredDirective } from '@energinet-datahub/dh/shared/feat
...
-    inset="ml"
-    *transloco="let t; read: 'admin.userManagement.permissionsTab'"
-  >
-    <vater-flex fill="vertical" gap="m">
-      <vater-stack direction="row" gap="s">
+  providers: [DhNavigationService],
+  template: `
+    <watt-card vater inset="ml" *transloco="let t; read: 'admin.userManagement.permissionsTab'">
+      <watt-data-table
+        vater
```
Comment (dzhavat) [inline libs/dh/admin/feature-permissions-new/src/lib/table.component.ts:66]: issue: the page looks like there are nested cards. The `watt-card` should probably be removed?

image

File: libs/dh/admin/feature-permissions-new/src/lib/download.component.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    </watt-button>
+
+    <watt-button
+      *dhPermissionRequired="['user-roles:manage']"
+      icon="download"
+      variant="text"
+      (click)="downloadRelationCSV(url())"
+    >
+      {{ 'shared.downloadreport' | transloco }}
+    </watt-button>
```
Comment (dzhavat) [inline libs/dh/admin/feature-permissions-new/src/lib/download.component.ts:46]: suggestion: let's give these two some `gap`

image

File: libs/dh/admin/feature-permissions-new/src/lib/table.component.ts (hunk 3 above)
Comment (mimse) [reply libs/dh/admin/feature-permissions-new/src/lib/table.component.ts:66]: You are right, data table has its own one

**Review by dzhavat — APPROVED** [review]

**Pull request merged**